### ISSUE TITLE
Xrm.WebAPI.retrieveRecord function has wrong parameter type

### ISSUE THREAD

**Issue by DavidPezzoli** [delegateas/XrmDefinitelyTyped#82]
Hi,
Why not also provide the classic Microsoft call? In my opinion it is more understandable to many

Also The parameters of this API are incorrect into your template library:
Xrm.WebApi.createRecord(entityLogicalName, data).then(successCallback, errorCallback);

Thanks,
David

**Comment by magesoe** [comment]
You're right, we will have to fix those typings.

However, it seems to me that the standard Microsoft call uses JSON. We like strongly typed queries. Therefore we have created `XrmQuery`, which is located in /lib/xx.xrmquery.web.js and has typings in the typing folder. If you add that js to your form, then you can use it.

I can see our current wiki is sparse in regards to what XrmQuery can do, but you can create a record like this.
```typescript
XrmQuery.create(x => x.accounts)
  .setData({
    name: "Some account name",
    address1_line1: "Great street"
  })
  .execute(id => console.log("Success! Created with id: " + id),
    err => console.log("Error: " + err.message))
```
All of this is strongly typed, so you can only add attributes that exists on an account. However, Typescript also has async/await, so instead of using callbacks you can do the following.
```typescript
let id = await XrmQuery.create(x => x.accounts)
  .setData({
    name: "Some account name",
    address1_line1: "Great street"
  }).promise();

// using id as a normal variable
console.log("Success! " + id);
```
This helps keeping the code clean instead of nesting `.then` calls, like you would also have to do with the Microsoft call.

**Comment by DavidPezzoli** [comment]
Hi Magnus Gether Sørensen,
thank you for your reply.
Your library is fantastic! But many CRM developers will bother you for the Xrm.WebAPI. In order to get the code more readable, some CRM developer like me, would like to use directly the Xrm.WebAPI provided by Microsoft.
However everything it's OK, the input parameters Xrm.WebAPI are JSON JavaScript.

Thanks again!

**Comment by magesoe** [comment]
That is alright, but I would still argue that one should try to move away from using untyped code, such as passing a JSON structure to Xrm.WebAPI. None the less, our typings are incorrect. You're welcome to make a pull request, otherwise I will fix it when I get the time :)

**Comment by magesoe** [comment]
I could not find any errors regarding retrieveRecords, but create and update had errors in the typings and is fixed in version 4.0.9
